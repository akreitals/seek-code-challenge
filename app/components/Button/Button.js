import styled from 'styled-components';
import PropTypes from 'prop-types';
import { space } from 'styled-system';
import theme from '../theme';

const fullWidth = props => (props.fullWidth ? { width: '100%' } : null);

const Button = styled('button')`
    -webkit-font-smoothing: antialiased;
    display: inline-block;
    vertical-align: middle;
    text-align: center;
    text-decoration: none;
    font-family: inherit;
    font-weight: ${theme.regular};
    line-height: ${theme.fontSizes[6]}px;
    cursor: pointer;
    border-radius: ${props => props.theme.radius};
    background-color: ${props => props.theme.colors.primary};
    color: ${props => props.theme.colors.white};
    border-width: 0;
    border-style: solid;
    font-size: ${theme.fontSizes[4]}px;
    padding: 0 ${theme.space[4]}px;

    ${props => props.disabled && { opacity: 0.25, cursor: 'not-allowed' }};

    &:hover {
        background-color: ${props =>
            props.disabled ? null : props.theme.colors.darkBlue};
    }

    &:focus {
        outline: none;
        z-index: 1;
        box-shadow: 0 0 0 2px rgba(30, 144, 255, 0.75);

        &::-moz-focus-inner {
            border: 0;
        }
    }

    ${fullWidth} ${space};
`;

Button.displayName = 'Button';

Button.propTypes = {
    fullWidth: PropTypes.bool,
    disabled: PropTypes.bool
};

Button.defaultProps = {
    theme
};

Button.input = Button.withComponent('input');

export default Button;
