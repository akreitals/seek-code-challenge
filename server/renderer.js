import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
