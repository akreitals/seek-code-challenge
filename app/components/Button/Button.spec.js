import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Button from './Button';

describe('Button', () => {
    test('should render a <Button /> component', () => {
        const json = renderer.create(<Button />).toJSON();
        expect(json).toMatchSnapshot();
    });

    test('should render a disabled button when disabled prop is set', () => {
        const json = renderer.create(<Button disabled />).toJSON();
        expect(json).toMatchSnapshot();
        expect(json).toHaveStyleRule('opacity', '0.25');
        expect(json).toHaveStyleRule('cursor', 'not-allowed');
    });
});
