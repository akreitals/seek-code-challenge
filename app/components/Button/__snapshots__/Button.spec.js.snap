// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a <Button /> component 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 400;
  line-height: 42px;
  cursor: pointer;
  background-color: #e60278;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  font-size: 24px;
  padding: 0 32px;
}

.c0:focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 2px rgba(30,144,255,0.75);
}

.c0:focus::-moz-focus-inner {
  border: 0;
}

<button
  className="c0"
/>
`;

exports[`Button should render a disabled button when disabled prop is set 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 400;
  line-height: 42px;
  cursor: pointer;
  background-color: #e60278;
  color: #ffffff;
  border-width: 0;
  border-style: solid;
  font-size: 24px;
  padding: 0 32px;
  opacity: 0.25;
  cursor: not-allowed;
}

.c0:focus {
  outline: none;
  z-index: 1;
  box-shadow: 0 0 0 2px rgba(30,144,255,0.75);
}

.c0:focus::-moz-focus-inner {
  border: 0;
}

<button
  className="c0"
  disabled={true}
/>
`;
